packages/core/test/state.test.ts

```diff
@@ -615,6 +615,41 @@ describe('State', () => {
       expect(machine.initialState.can({ type: 'NEXT' })).toBe(true);
     });
 
+    it('should return true for an event object that results in a new action', () => {
+      const machine = createMachine({
+        initial: 'a',
+        states: {
+          a: {
+            on: {
+              NEXT: {
+                actions: 'newAction'
+              }
+            }
+          }
+        }
+      });
+
+      expect(machine.initialState.can({ type: 'NEXT' })).toBe(true);
+    });
```

Comment:
note - this is basically a duplicate of https://github.com/statelyai/xstate/blob/65c7e62a506d34648fb046278ec0427943236981/packages/core/test/state.test.ts#L702-L717